src/uu/cksum/src/cksum.rs

```diff
@@ -439,20 +449,53 @@ where
         };
         let reader = BufReader::new(file);
 
+        // for each line in the input, check if it is a valid checksum line
         for line in reader.lines() {
-            let line = line?;
-
+            let line = line.unwrap_or_else(|_| String::new());
             if let Some(caps) = re.captures(&line) {
                 properly_formatted = true;
-                let algo_name = caps.name("algo").unwrap().as_str().to_lowercase();
-                let filename_to_check = caps.name("filename").unwrap().as_str();
-                let expected_checksum = caps.name("checksum").unwrap().as_str();
 
-                let length = caps
-                    .name("bits")
-                    .map(|m| m.as_str().parse::<usize>().unwrap() / 8);
-                let (_, mut algo, bits) = detect_algo(&algo_name, length);
+                // Determine what kind of file input we had
+                // we need it for case -"-check -a sm3 <file>" when <file> is
```

Comment:
Is it possible that you meant `"--check -a sm3 <file>"` , i.e. the `-` and `"` got swapped?